lib/twilio/device.ts

```diff
@@ -559,22 +557,45 @@ class Device extends EventEmitter {
    * @param options
    */
   async connect(options: Device.ConnectOptions = { }): Promise<Call> {
-    this._log.debug('.connect', JSON.stringify(options && options.params || {}), options);
+    this._log.debug('.connect', JSON.stringify(options));
     this._throwIfDestroyed();
-
     if (this._activeCall) {
       throw new InvalidStateError('A Call is already active');
     }
 
-    const activeCall = this._activeCall = await this._makeCall(
-      options.params || { },
-      {
-        enableImprovedSignalingErrorPrecision:
-          !!this._options.enableImprovedSignalingErrorPrecision,
-        rtcConfiguration: options.rtcConfiguration,
-        voiceEventSidGenerator: this._options.voiceEventSidGenerator,
-      },
-    );
+    let callSid, customParameters, parameters, signalingReconnectToken;
+    if (options.connectToken) {
+      try {
+        const parts = JSON.parse(atob(options.connectToken));
+        callSid = parts.callSid;
+        customParameters = parts.customParameters;
+        parameters = parts.parameters;
+        signalingReconnectToken = parts.signalingReconnectToken;
```

Comment:
Do you think we should add checks here for `typeof part === 'string' && part.length > 0`?

Reply:
Not 100% sure what you mean. Can you add the code snippet?

Reply:
```typescript
callSid = parts.callSid;
if (typeof callSid !== 'string' || callSid.length === 0) {
  throw...
}
```
For each of the parts.

Reply:
Ok. I added validation only for the required values. Also, I'm only checking for truthy values instead of checking every single type and length for each parameter. Keeps the code a bit simpler. If the user re-encodes the connectToken with bad types which is highly unlikely, we fail anyway on the server.